diff -up chromium-20090909svn25824/src/webkit/glue/plugins/plugin_list_linux.cc.BAD chromium-20090909svn25824/src/webkit/glue/plugins/plugin_list_linux.cc
--- chromium-20090909svn25824/src/webkit/glue/plugins/plugin_list_linux.cc.BAD	2009-09-09 21:16:07.000000000 -0400
+++ chromium-20090909svn25824/src/webkit/glue/plugins/plugin_list_linux.cc	2009-09-09 21:17:05.000000000 -0400
@@ -86,9 +86,16 @@ void PluginList::GetPluginDirectories(st
 
   // 3) NS_SYSTEM_PLUGINS_DIR:
   // This varies across different versions of Firefox, so check 'em all.
+  // TODO: add additional supported 64-bit architectures here
+  #if defined(_x86_64_)
+  plugin_dirs->push_back(FilePath("/usr/lib64/mozilla/plugins"));
+  plugin_dirs->push_back(FilePath("/usr/lib64/firefox/plugins"));
+  plugin_dirs->push_back(FilePath("/usr/lib64/xulrunner-addons/plugins"));
+  #else
   plugin_dirs->push_back(FilePath("/usr/lib/mozilla/plugins"));
   plugin_dirs->push_back(FilePath("/usr/lib/firefox/plugins"));
   plugin_dirs->push_back(FilePath("/usr/lib/xulrunner-addons/plugins"));
+  #endif
 }
 
 void PluginList::LoadPluginsFromDir(const FilePath& path,
