--- chromium-browser-14.0.783.0~svn20110602r87572/src/third_party/tcmalloc/chromium/src/tcmalloc.cc.org	2011-06-02 19:20:28.041396841 +0200
+++ chromium-browser-14.0.783.0~svn20110602r87572/src/third_party/tcmalloc/chromium/src/tcmalloc.cc	2011-06-02 19:20:42.624730752 +0200
@@ -375,7 +375,7 @@
   return tc_malloc(size);
 }
 
-void* (*__malloc_hook)(
+void* (* volatile __malloc_hook)(
     size_t size, const void* caller) = tc_ptmalloc_malloc_hook;
 
 static void* tc_ptmalloc_realloc_hook(
@@ -383,14 +383,14 @@
   return tc_realloc(ptr, size);
 }
 
-void* (*__realloc_hook)(
+void* (* volatile __realloc_hook)(
     void* ptr, size_t size, const void* caller) = tc_ptmalloc_realloc_hook;
 
 static void tc_ptmalloc_free_hook(void* ptr, const void* caller) {
   tc_free(ptr);
 }
 
-void (*__free_hook)(void* ptr, const void* caller) = tc_ptmalloc_free_hook;
+void (* volatile __free_hook)(void* ptr, const void* caller) = tc_ptmalloc_free_hook;
 
 #endif
 
@@ -1584,7 +1584,7 @@
   MallocHook::InvokeNewHook(result, size);
   return result;
 }
-void *(*__memalign_hook)(size_t, size_t, const void *) = MemalignOverride;
+void *(* volatile __memalign_hook)(size_t, size_t, const void *) = MemalignOverride;
 #endif  // TCMALLOC_USING_DEBUGALLOCATION
 
 // ---Double free() debugging implementation -----------------------------------
