diff -up chromium-20090916svn26424/src/base/debug_util_posix.cc.BAD chromium-20090916svn26424/src/base/debug_util_posix.cc
--- chromium-20090916svn26424/src/base/debug_util_posix.cc.BAD	2009-09-16 22:57:39.000000000 -0400
+++ chromium-20090916svn26424/src/base/debug_util_posix.cc	2009-09-16 22:57:46.000000000 -0400
@@ -116,9 +116,13 @@ void DebugUtil::BreakDebugger() {
 }
 
 StackTrace::StackTrace() {
+#if defined(OS_MACOSX)
   if (backtrace == NULL) {
     count_ = 0;
   } else {
+#else
+  {
+#endif
     // Though the backtrace API man page does not list any possible negative
     // return values, we take no chance.
     count_ = std::max(backtrace(trace_, arraysize(trace_)), 0);
@@ -126,14 +130,22 @@ StackTrace::StackTrace() {
 }
 
 void StackTrace::PrintBacktrace() {
+#if defined(OS_MACOSX)
   if (backtrace_symbols_fd != NULL) {
+#else
+  {
+#endif
     fflush(stderr);
     backtrace_symbols_fd(trace_, count_, STDERR_FILENO);
   }
 }
 
 void StackTrace::OutputToStream(std::ostream* os) {
+#if defined(OS_MACOSX)
   if (backtrace_symbols != NULL) {
+#else
+  {
+#endif
     scoped_ptr_malloc<char*> trace_symbols(backtrace_symbols(trace_, count_));
 
     // If we can't retrieve the symbols, print an error and just dump the raw
