--- chromium-browser-34.0.1847.116/build/gyp_chromium~	2014-04-02 22:03:54.000000000 +0300
+++ chromium-browser-34.0.1847.116/build/gyp_chromium	2014-04-12 01:46:51.740436678 +0300
@@ -477,8 +477,7 @@
     args.append('--check')
 
   supplemental_includes = GetSupplementalFiles()
-  if not RunGN(supplemental_includes):
-    sys.exit(1)
+
   args.extend(
       ['-I' + i for i in additional_include_files(supplemental_includes, args)])
 
