--- chromium-browser-33.0.1750.117/media/audio/alsa/audio_manager_alsa.cc~	2014-02-20 22:27:58.000000000 +0200
+++ chromium-browser-33.0.1750.117/media/audio/alsa/audio_manager_alsa.cc	2014-02-23 10:24:29.610111342 +0200
@@ -53,7 +53,8 @@
   CommandLine command_line(CommandLine::NO_PROGRAM);
   switch (base::nix::GetDesktopEnvironment(env.get())) {
     case base::nix::DESKTOP_ENVIRONMENT_GNOME:
-      command_line.SetProgram(base::FilePath("gnome-volume-control"));
+      command_line.SetProgram(base::FilePath("gnome-control-center"));
+      command_line.AppendArg("sound");
       break;
     case base::nix::DESKTOP_ENVIRONMENT_KDE3:
     case base::nix::DESKTOP_ENVIRONMENT_KDE4:
