--- chromium-browser-50.0.2661.102/media/audio/alsa/audio_manager_alsa.cc.orig	2021-05-22 17:02:40.236005735 +0200
+++ chromium-browser-50.0.2661.102/media/audio/alsa/audio_manager_alsa.cc	2021-05-22 17:07:10.666708626 +0200
@@ -56,7 +56,8 @@ void AudioManagerAlsa::ShowLinuxAudioInp
   base::CommandLine command_line(base::CommandLine::NO_PROGRAM);
   switch (base::nix::GetDesktopEnvironment(env.get())) {
     case base::nix::DESKTOP_ENVIRONMENT_GNOME:
-      command_line.SetProgram(base::FilePath("gnome-volume-control"));
+      command_line.SetProgram(base::FilePath("gnome-control-center"));
+      command_line.AppendArg("sound");
       break;
     case base::nix::DESKTOP_ENVIRONMENT_KDE3:
     case base::nix::DESKTOP_ENVIRONMENT_KDE4:
