--- chromium-browser-26.0.1410.12/src/media/audio/linux/audio_manager_linux.cc~	2013-02-21 12:02:26.000000000 +0200
+++ chromium-browser-26.0.1410.12/src/media/audio/linux/audio_manager_linux.cc	2013-02-25 15:16:54.634665398 +0200
@@ -92,7 +92,8 @@
   CommandLine command_line(CommandLine::NO_PROGRAM);
   switch (base::nix::GetDesktopEnvironment(env.get())) {
     case base::nix::DESKTOP_ENVIRONMENT_GNOME:
-      command_line.SetProgram(base::FilePath("gnome-volume-control"));
+      command_line.SetProgram(base::FilePath("gnome-control-center"));
+      command_line.AppendArg("sound");
       break;
     case base::nix::DESKTOP_ENVIRONMENT_KDE3:
     case base::nix::DESKTOP_ENVIRONMENT_KDE4:
