--- chromium-browser-16.0.912.21~r108057/src/third_party/zlib/zlib.gyp~	2011-11-05 15:05:40.000000000 +0100
+++ chromium-browser-16.0.912.21~r108057/src/third_party/zlib/zlib.gyp	2011-11-05 17:31:17.605374715 +0100
@@ -88,12 +88,6 @@
             'USE_SYSTEM_ZLIB',
           ],
           'sources': [
-            'contrib/minizip/ioapi.c',
-            'contrib/minizip/ioapi.h',
-            'contrib/minizip/unzip.c',
-            'contrib/minizip/unzip.h',
-            'contrib/minizip/zip.c',
-            'contrib/minizip/zip.h',
           ],
           'conditions': [
             ['OS=="android"', {
@@ -102,7 +96,7 @@
           ],
           'link_settings': {
             'libraries': [
-              '-lz',
+              '-lminizip',
             ],
           },
         },
