diff -up chromium-20090916svn26424/src/chrome/browser/autocomplete/autocomplete_popup_model.cc.icu chromium-20090916svn26424/src/chrome/browser/autocomplete/autocomplete_popup_model.cc
--- chromium-20090916svn26424/src/chrome/browser/autocomplete/autocomplete_popup_model.cc.icu	2009-09-16 22:03:33.000000000 -0400
+++ chromium-20090916svn26424/src/chrome/browser/autocomplete/autocomplete_popup_model.cc	2009-09-16 22:05:14.000000000 -0400
@@ -11,7 +11,7 @@
 #include "chrome/browser/search_engines/template_url.h"
 #include "chrome/browser/search_engines/template_url_model.h"
 #include "chrome/common/notification_service.h"
-#include "third_party/icu/public/common/unicode/ubidi.h"
+#include <unicode/ubidi.h>
 
 ///////////////////////////////////////////////////////////////////////////////
 // AutocompletePopupModel
diff -up chromium-20090916svn26424/src/chrome/browser/spellcheck_worditerator.cc.icu chromium-20090916svn26424/src/chrome/browser/spellcheck_worditerator.cc
--- chromium-20090916svn26424/src/chrome/browser/spellcheck_worditerator.cc.icu	2009-09-16 22:14:29.000000000 -0400
+++ chromium-20090916svn26424/src/chrome/browser/spellcheck_worditerator.cc	2009-09-16 22:16:22.000000000 -0400
@@ -11,12 +11,12 @@
 #include "base/string_util.h"
 #include "chrome/browser/spellchecker.h"
 
-#include "third_party/icu/public/common/unicode/normlzr.h"
-#include "third_party/icu/public/common/unicode/schriter.h"
-#include "third_party/icu/public/common/unicode/uchar.h"
-#include "third_party/icu/public/common/unicode/uscript.h"
-#include "third_party/icu/public/common/unicode/uset.h"
-#include "third_party/icu/public/i18n/unicode/ulocdata.h"
+#include <unicode/normlzr.h>
+#include <unicode/schriter.h>
+#include <unicode/uchar.h>
+#include <unicode/uscript.h>
+#include <unicode/uset.h>
+#include <unicode/ulocdata.h>
 
 SpellcheckCharAttribute::SpellcheckCharAttribute() {
   InitializeScriptTable();
