--- chromium-browser-20.0.1132.47/native_client/SConstruct~	2012-06-28 16:03:20.000000000 +0300
+++ chromium-browser-20.0.1132.47/native_client/SConstruct	2012-07-07 01:27:46.054400799 +0300
@@ -3329,7 +3329,7 @@
     #### ALPHABETICALLY SORTED ####
     ], ppapi_scons_files['nonvariant_test_scons_files'])
 
-nacl_env.Append(BUILD_SCONSCRIPTS=irt_variant_tests + nonvariant_tests)
+#nacl_env.Append(BUILD_SCONSCRIPTS=irt_variant_tests + nonvariant_tests)
 
 # ----------------------------------------------------------
 # Possibly install an sdk by downloading it
