--- chromium-browser-24.0.1312.14/./src/native_client/build/untrusted.gypi~	2012-11-14 04:37:51.000000000 +0200
+++ chromium-browser-24.0.1312.14/./src/native_client/build/untrusted.gypi	2012-11-16 22:01:20.476042441 +0200
@@ -162,7 +162,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-64',
                    '--build', 'newlib_nexe',
@@ -203,7 +203,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-64',
                    '--build', 'newlib_nlib',
@@ -244,7 +244,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-32',
                    '--build', 'newlib_nexe',
@@ -285,7 +285,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-32',
                    '--build', 'newlib_nlib',
@@ -331,7 +331,7 @@
                 'action': [
                   '>(python_exe)',
                   '<(DEPTH)/native_client/build/build_nexe.py',
-                  '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                  '-t', '<(DEPTH)/native_client/toolchain/',
                   '>@(extra_args)',
                   '--arch', 'arm',
                   '--build', 'newlib_nexe',
@@ -371,7 +371,7 @@
                 'action': [
                   '>(python_exe)',
                   '<(DEPTH)/native_client/build/build_nexe.py',
-                  '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                  '-t', '<(DEPTH)/native_client/toolchain/',
                   '>@(extra_args)',
                   '--arch', 'arm',
                   '--build', 'newlib_nlib',
@@ -418,7 +418,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-64',
                    '--build', 'glibc_nexe',
@@ -459,7 +459,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-32',
                    '--build', 'glibc_nexe',
@@ -500,7 +500,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-64',
                    '--build', 'glibc_nlib',
@@ -541,7 +541,7 @@
                  'action': [
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '>@(extra_args)',
                    '--arch', 'x86-32',
                    '--build', 'glibc_nlib',
@@ -583,7 +583,7 @@
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
                    '>@(extra_args)',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '--arch', 'x86-64',
                    '--build', 'glibc_nso',
                    '--root', '<(DEPTH)',
@@ -624,7 +624,7 @@
                    '>(python_exe)',
                    '<(DEPTH)/native_client/build/build_nexe.py',
                    '>@(extra_args)',
-                   '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+                   '-t', '<(DEPTH)/native_client/toolchain/',
                    '--arch', 'x86-32',
                    '--build', 'glibc_nso',
                    '--root', '<(DEPTH)',
@@ -702,7 +702,7 @@
             'action': [
               '>(python_exe)',
               '<(DEPTH)/native_client/build/build_nexe.py',
-              '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+              '-t', '<(DEPTH)/native_client/toolchain/',
               '>@(extra_args)',
               '--arch', 'pnacl',
               '--build', 'newlib_pexe',
@@ -743,7 +743,7 @@
             'action': [
               '>(python_exe)',
               '<(DEPTH)/native_client/build/build_nexe.py',
-              '-t', '<(SHARED_INTERMEDIATE_DIR)/sdk/toolchain/',
+              '-t', '<(DEPTH)/native_client/toolchain/',
               '>@(extra_args)',
               '--arch', 'pnacl',
               '--build', 'newlib_plib',
