--- chromium-browser-25.0.1364.5/src/native_client/build/untrusted.gypi~	2013-01-05 22:28:19.000000000 +0200
+++ chromium-browser-25.0.1364.5/src/native_client/build/untrusted.gypi	2013-01-05 22:29:09.503508356 +0200
@@ -67,7 +67,7 @@
           'build_newlib': 0,
           'build_glibc': 0,
           'disable_glibc%': 0,
-          'extra_args': [],
+          'extra_args': ['--verbose'],
           'enable_x86_32': 1,
           'enable_x86_64': 1,
           'extra_deps_newlib64': [],
@@ -116,7 +116,7 @@
           'nacl_enable_arm_gcc%': 0,
           'build_glibc': 0,
           'disable_glibc%': 1,
-          'extra_args': [],
+          'extra_args': ['--verbose'],
           'enable_x86_32': 0,
           'enable_x86_64': 0,
           'enable_arm': 1,
