--- chromium-browser-50.0.2661.102/native_client/src/trusted/service_runtime/service_runtime.gyp.orig	2021-05-22 14:30:20.134879142 +0200
+++ chromium-browser-50.0.2661.102/native_client/src/trusted/service_runtime/service_runtime.gyp	2021-05-22 14:55:16.892591353 +0200
@@ -113,6 +113,16 @@
                 'win/thread_handle_map.c',
               ],
             }],
+            ['target_arch=="ia32"', {
+              'sources': [
+                'linux/nacl_signal_32.c',
+              ],
+            }],
+            ['target_arch=="x64"', {
+              'sources': [
+                'linux/nacl_signal_64.c',
+              ],
+            }],
             # TODO(gregoryd): move arm-specific stuff into a separate gyp file.
             ['target_arch=="arm"', {
               'sources': [
