--- chromium-browser-5.0.382.0~svn20100418r44898/src/build/common.gypi~	2010-04-19 12:58:15.000000000 +0300
+++ chromium-browser-5.0.382.0~svn20100418r44898/src/build/common.gypi	2010-04-19 13:02:22.913765194 +0300
@@ -199,6 +199,18 @@
     # Set this to true to enable SELinux support.
     'selinux%': 0,
 
+    # Use system libpng
+    'use_system_libpng%': 1,
+
+    # Use system bzip2
+    'use_system_bzip2%': 1,
+
+    # Use system libjpeg
+    'use_system_libjpeg%': 1,
+
+    # Use system zlib
+    'use_system_zlib%': 1,
+
     # Strip the binary after dumping symbols.
     'linux_strip_binary%': 0,
 
@@ -215,6 +227,18 @@
       ['selinux==1', {
         'defines': ['CHROMIUM_SELINUX=1'],
       }],
+      ['use_system_libpng==1', {
+        'defines': ['USE_SYSTEM_LIBPNG=1'],
+      }],
+      ['use_system_bzip2==1', {
+        'defines': ['USE_SYSTEM_LIBBZ2=1'],
+      }],
+      ['use_system_libjpeg==1', {
+        'defines': ['USE_SYSTEM_LIBJPEG=1'],
+      }],
+      ['use_system_zlib==1', {
+        'defines': ['USE_SYSTEM_ZLIB=1'],
+      }],
       ['coverage!=0', {
         'conditions': [
           ['OS=="mac"', {
