--- chromium-browser-7.0.505.0~svn20100825r57278/src/build/common.gypi~	2010-08-25 10:57:49.000000000 +0300
+++ chromium-browser-7.0.505.0~svn20100825r57278/src/build/common.gypi	2010-08-25 10:59:07.552987859 +0300
@@ -249,6 +249,18 @@
     # Set this to true to enable SELinux support.
     'selinux%': 0,
 
+    # Use system libpng
+    'use_system_libpng%': 1,
+
+    # Use system bzip2
+    'use_system_bzip2%': 1,
+
+    # Use system libjpeg
+    'use_system_libjpeg%': 1,
+
+    # Use system zlib
+    'use_system_zlib%': 1,
+
     # Override whether we should use Breakpad on Linux. I.e. for Chrome bot.
     'linux_breakpad%': 0,
     # And if we want to dump symbols for Breakpad-enabled builds.
@@ -215,6 +227,18 @@
       ['selinux==1', {
         'defines': ['CHROMIUM_SELINUX=1'],
       }],
+      ['use_system_libpng==1', {
+        'defines': ['USE_SYSTEM_LIBPNG=1'],
+      }],
+      ['use_system_bzip2==1', {
+        'defines': ['USE_SYSTEM_LIBBZ2=1'],
+      }],
+      ['use_system_libjpeg==1', {
+        'defines': ['USE_SYSTEM_LIBJPEG=1'],
+      }],
+      ['use_system_zlib==1', {
+        'defines': ['USE_SYSTEM_ZLIB=1'],
+      }],
       ['coverage!=0', {
         'conditions': [
           ['OS=="mac"', {
