--- chromium-browser-5.0.313.0~svn20100202r37790/src/webkit/glue/plugins/plugin_list_posix.cc~	2010-02-02 15:14:25.933074719 +0200
+++ chromium-browser-5.0.313.0~svn20100202r37790/src/webkit/glue/plugins/plugin_list_posix.cc	2010-02-02 16:08:06.014003355 +0200
@@ -90,6 +86,8 @@
   if (home)
     plugin_dirs->push_back(FilePath(home).Append(".mozilla/plugins"));
 
+  // in PLD Linux we want to search only Chrome specific dirs.
+#if 0
   // 3) NS_SYSTEM_PLUGINS_DIR:
   // This varies across different browsers and versions, so check 'em all.
   plugin_dirs->push_back(FilePath("/usr/lib/browser-plugins"));
@@ -101,6 +105,7 @@
   plugin_dirs->push_back(FilePath("/usr/lib64/firefox/plugins"));
   plugin_dirs->push_back(FilePath("/usr/lib64/xulrunner-addons/plugins"));
 #endif
+#endif
 }
 
 void PluginList::LoadPluginsFromDir(const FilePath& path,
