--- chromium-browser-10.0.632.0~svn20110107r70714/src/webkit/plugins/npapi/plugin_list_posix.cc~	2010-12-22 04:02:37.000000000 +0100
+++ chromium-browser-10.0.632.0~svn20110107r70714/src/webkit/plugins/npapi/plugin_list_posix.cc	2011-01-07 17:41:38.193692743 +0100
@@ -146,6 +146,8 @@
   if (!home.empty())
     plugin_dirs->push_back(home.Append(".mozilla/plugins"));
 
+  // in PLD Linux we want to search only Chrome specific dirs.
+#if 0
   // 3) NS_SYSTEM_PLUGINS_DIR:
   // This varies across different browsers and versions, so check 'em all.
   plugin_dirs->push_back(FilePath("/usr/lib/browser-plugins"));
@@ -161,6 +163,7 @@
   plugin_dirs->push_back(FilePath("/usr/lib64/firefox/plugins"));
   plugin_dirs->push_back(FilePath("/usr/lib64/xulrunner-addons/plugins"));
 #endif  // defined(ARCH_CPU_64_BITS)
+#endif // PLD Linux
 #endif  // !defined(OS_CHROMEOS)
 }
 
