--- chromium-browser-7.0.509.0~svn20100830r57827/src/webkit/glue/plugins/plugin_list_posix.cc~	2010-08-30 13:11:06.000000000 +0300
+++ chromium-browser-7.0.509.0~svn20100830r57827/src/webkit/glue/plugins/plugin_list_posix.cc	2010-08-30 14:34:22.711621299 +0300
@@ -90,6 +86,8 @@
   if (home)
     plugin_dirs->push_back(FilePath(home).Append(".mozilla/plugins"));
 
+  // in PLD Linux we want to search only Chrome specific dirs.
+#if 0
   // 3) NS_SYSTEM_PLUGINS_DIR:
   // This varies across different browsers and versions, so check 'em all.
   plugin_dirs->push_back(FilePath("/usr/lib/browser-plugins"));
@@ -161,6 +161,7 @@
   plugin_dirs->push_back(FilePath("/usr/lib64/firefox/plugins"));
   plugin_dirs->push_back(FilePath("/usr/lib64/xulrunner-addons/plugins"));
 #endif  // defined(ARCH_CPU_64_BITS)
+#endif // PLD Linux
 #endif  // !defined(OS_CHROMEOS)
 }
 
