http://crbug.com/174287
--- branches/1410/src/media/media.gyp	2013/03/05 22:36:07	186272
+++ branches/1410/src/media/media.gyp	2013/03/05 22:37:51	186273
@@ -12,12 +12,12 @@
     'conditions': [
       ['OS == "android" or OS == "ios"', {
         # Android and iOS don't use ffmpeg.
-        'use_ffmpeg%': 0,
+        'media_use_ffmpeg%': 0,
         # Android and iOS don't use libvpx.
-        'use_libvpx%': 0,
+        'media_use_libvpx%': 0,
       }, {  # 'OS != "android" and OS != "ios"'
-        'use_ffmpeg%': 1,
-        'use_libvpx%': 1,
+        'media_use_ffmpeg%': 1,
+        'media_use_libvpx%': 1,
       }],
       # Screen capturer works only on Windows, OSX and Linux.
       ['OS=="win" or OS=="mac" or OS=="linux"', {
@@ -426,14 +426,14 @@
             'yuv_convert',
           ],
         }],
-        ['use_ffmpeg == 1', {
+        ['media_use_ffmpeg == 1', {
           'dependencies': [
             '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
           ],
           'export_dependent_settings': [
             '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
           ],
-        }, {  # use_ffmpeg == 0
+        }, {  # media_use_ffmpeg == 0
           # Exclude the sources that depend on ffmpeg.
           'sources!': [
             'base/media_posix.cc',
@@ -458,11 +458,16 @@
             'webm/webm_stream_parser.h',
           ],
         }],
-        ['use_libvpx == 1', {
+        ['media_use_libvpx == 1', {
           'dependencies': [
             '<(DEPTH)/third_party/libvpx/libvpx.gyp:libvpx',
           ],
-        }, {  # use_libvpx == 0
+        }, {  # media_use_libvpx == 0
+          'direct_dependent_settings': {
+            'defines': [
+              'MEDIA_DISABLE_LIBVPX',
+            ],
+          },
           # Exclude the sources that depend on libvpx.
           'sources!': [
             'filters/vpx_video_decoder.cc',
@@ -853,7 +858,7 @@
             'yuv_convert',
           ],
         }],
-        ['use_ffmpeg == 1', {
+        ['media_use_ffmpeg == 1', {
           'dependencies': [
             '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
           ],
--- branches/1410/src/webkit/media/filter_helpers.cc	2013/03/05 22:36:07	186272
+++ branches/1410/src/webkit/media/filter_helpers.cc	2013/03/05 22:37:51	186273
@@ -45,11 +45,15 @@
       new media::FFmpegVideoDecoder(message_loop);
   filter_collection->GetVideoDecoders()->push_back(ffmpeg_video_decoder);
 
+  // TODO(phajdan.jr): Remove ifdefs when libvpx with vp9 support is released
+  // (http://crbug.com/174287) .
+#if !defined(MEDIA_DISABLE_LIBVPX)
   if (cmd_line->HasSwitch(switches::kEnableVp9Playback)) {
     scoped_refptr<media::VpxVideoDecoder> vpx_video_decoder =
         new media::VpxVideoDecoder(message_loop);
     filter_collection->GetVideoDecoders()->push_back(vpx_video_decoder);
   }
+#endif  // defined(MEDIA_USE_LIBVPX)
 }
 
 bool BuildMediaStreamCollection(
--- branches/1410/src/webkit/media/webkit_media.gypi	2013/03/05 22:36:07	186272
+++ branches/1410/src/webkit/media/webkit_media.gypi	2013/03/05 22:37:51	186273
@@ -34,6 +34,7 @@
       'dependencies': [
         '<(DEPTH)/base/base.gyp:base',
         '<(DEPTH)/base/third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
+        '<(DEPTH)/media/media.gyp:media',
         '<(DEPTH)/media/media.gyp:shared_memory_support',
         '<(DEPTH)/media/media.gyp:yuv_convert',
         '<(DEPTH)/skia/skia.gyp:skia',
