diff -up chromium-20090827svn24640/src/base/message_pump_libevent.cc.system-code chromium-20090827svn24640/src/base/message_pump_libevent.cc
--- chromium-20090827svn24640/src/base/message_pump_libevent.cc.system-code	2009-08-27 15:21:27.000000000 -0400
+++ chromium-20090827svn24640/src/base/message_pump_libevent.cc	2009-08-27 16:05:17.000000000 -0400
@@ -12,7 +12,7 @@
 #include "base/scoped_nsautorelease_pool.h"
 #include "base/scoped_ptr.h"
 #include "base/time.h"
-#include "third_party/libevent/event.h"
+#include <event.h>
 
 // Lifecycle of struct event
 // Libevent uses two main data structures:
diff -up chromium-20090827svn24640/src/base/pr_time_unittest.cc.system-code chromium-20090827svn24640/src/base/pr_time_unittest.cc
--- chromium-20090827svn24640/src/base/pr_time_unittest.cc.system-code	2009-08-27 15:21:27.000000000 -0400
+++ chromium-20090827svn24640/src/base/pr_time_unittest.cc	2009-08-27 16:05:17.000000000 -0400
@@ -4,7 +4,7 @@
 
 #include <time.h>
 
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
diff -up chromium-20090827svn24640/src/base/time.cc.system-code chromium-20090827svn24640/src/base/time.cc
--- chromium-20090827svn24640/src/base/time.cc.system-code	2009-08-27 15:21:28.000000000 -0400
+++ chromium-20090827svn24640/src/base/time.cc	2009-08-27 16:05:17.000000000 -0400
@@ -5,7 +5,7 @@
 #include "base/time.h"
 #include "base/string_util.h"
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 
 #include "base/logging.h"
 
diff -up chromium-20090827svn24640/src/chrome/browser/debugger/devtools_remote_listen_socket.cc.system-code chromium-20090827svn24640/src/chrome/browser/debugger/devtools_remote_listen_socket.cc
--- chromium-20090827svn24640/src/chrome/browser/debugger/devtools_remote_listen_socket.cc.system-code	2009-08-27 15:22:41.000000000 -0400
+++ chromium-20090827svn24640/src/chrome/browser/debugger/devtools_remote_listen_socket.cc	2009-08-27 16:05:17.000000000 -0400
@@ -18,7 +18,7 @@
 #include "base/message_loop.h"
 #include "base/message_pump_libevent.h"
 #include "net/base/net_errors.h"
-#include "third_party/libevent/event.h"
+#include <event.h>
 #endif
 
 #include "base/eintr_wrapper.h"
diff -up chromium-20090827svn24640/src/chrome/common/zip.cc.system-code chromium-20090827svn24640/src/chrome/common/zip.cc
--- chromium-20090827svn24640/src/chrome/common/zip.cc.system-code	2009-08-27 15:22:46.000000000 -0400
+++ chromium-20090827svn24640/src/chrome/common/zip.cc	2009-08-27 16:05:17.000000000 -0400
@@ -7,8 +7,8 @@
 #include "base/file_util.h"
 #include "base/string_util.h"
 #include "net/base/file_stream.h"
-#include "third_party/zlib/contrib/minizip/unzip.h"
-#include "third_party/zlib/contrib/minizip/zip.h"
+#include <minizip/unzip.h>
+#include <minizip/zip.h>
 #if defined(OS_WIN)
 #include "third_party/zlib/contrib/minizip/iowin32.h"
 #endif
diff -up chromium-20090827svn24640/src/net/base/listen_socket.cc.system-code chromium-20090827svn24640/src/net/base/listen_socket.cc
--- chromium-20090827svn24640/src/net/base/listen_socket.cc.system-code	2009-08-27 15:20:57.000000000 -0400
+++ chromium-20090827svn24640/src/net/base/listen_socket.cc	2009-08-27 16:05:17.000000000 -0400
@@ -13,7 +13,7 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include "net/base/net_errors.h"
-#include "third_party/libevent/event.h"
+#include <event.h>
 #endif
 
 #include "base/eintr_wrapper.h"
diff -up chromium-20090827svn24640/src/net/base/telnet_server.cc.system-code chromium-20090827svn24640/src/net/base/telnet_server.cc
--- chromium-20090827svn24640/src/net/base/telnet_server.cc.system-code	2009-08-27 15:20:57.000000000 -0400
+++ chromium-20090827svn24640/src/net/base/telnet_server.cc	2009-08-27 16:05:17.000000000 -0400
@@ -13,7 +13,7 @@
 #include <sys/socket.h>
 #include "base/message_loop.h"
 #include "net/base/net_errors.h"
-#include "third_party/libevent/event.h"
+#include <event.h>
 #include "base/message_pump_libevent.h"
 #endif
 
diff -up chromium-20090827svn24640/src/net/socket/tcp_client_socket_libevent.cc.system-code chromium-20090827svn24640/src/net/socket/tcp_client_socket_libevent.cc
--- chromium-20090827svn24640/src/net/socket/tcp_client_socket_libevent.cc.system-code	2009-08-27 15:20:55.000000000 -0400
+++ chromium-20090827svn24640/src/net/socket/tcp_client_socket_libevent.cc	2009-08-27 16:05:17.000000000 -0400
@@ -15,7 +15,7 @@
 #include "base/trace_event.h"
 #include "net/base/io_buffer.h"
 #include "net/base/net_errors.h"
-#include "third_party/libevent/event.h"
+#include <event.h>
 
 
 namespace net {
diff -up chromium-20090827svn24640/src/webkit/glue/plugins/plugin_lib_linux.cc.system-code chromium-20090827svn24640/src/webkit/glue/plugins/plugin_lib_linux.cc
--- chromium-20090827svn24640/src/webkit/glue/plugins/plugin_lib_linux.cc.system-code	2009-08-27 15:19:05.000000000 -0400
+++ chromium-20090827svn24640/src/webkit/glue/plugins/plugin_lib_linux.cc	2009-08-27 16:10:57.000000000 -0400
@@ -13,7 +13,7 @@
 
 // These headers must be included in this order to make the declaration gods
 // happy.
-#include "base/third_party/nspr/prcpucfg_linux.h"
+#include <prcpucfg.h>
 #include "third_party/mozilla/include/nsplugindefs.h"
 
 namespace {
