diff -up chromium-20090827svn24640/src/base/pr_time_unittest.cc.system-code chromium-20090827svn24640/src/base/pr_time_unittest.cc
--- chromium-20090827svn24640/src/base/pr_time_unittest.cc.system-code	2009-08-27 15:21:27.000000000 -0400
+++ chromium-20090827svn24640/src/base/pr_time_unittest.cc	2009-08-27 16:05:17.000000000 -0400
@@ -4,7 +4,7 @@
 
 #include <time.h>
 
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "base/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
diff -up chromium-20090827svn24640/src/base/time.cc.system-code chromium-20090827svn24640/src/base/time.cc
--- chromium-20090827svn24640/src/base/time.cc.system-code	2009-08-27 15:21:28.000000000 -0400
+++ chromium-20090827svn24640/src/base/time.cc	2009-08-27 16:05:17.000000000 -0400
@@ -5,7 +5,7 @@
 #include "base/time.h"
 #include "base/string_util.h"
 #include "base/sys_string_conversions.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 
 #include "base/logging.h"
 
diff -up chromium-20090827svn24640/src/chrome/common/zip.cc.system-code chromium-20090827svn24640/src/chrome/common/zip.cc
--- chromium-20090827svn24640/src/chrome/common/zip.cc.system-code	2009-08-27 15:22:46.000000000 -0400
+++ chromium-20090827svn24640/src/chrome/common/zip.cc	2009-08-27 16:05:17.000000000 -0400
@@ -7,8 +7,8 @@
 #include "base/file_util.h"
 #include "base/string_util.h"
 #include "net/base/file_stream.h"
-#include "third_party/zlib/contrib/minizip/unzip.h"
-#include "third_party/zlib/contrib/minizip/zip.h"
+#include <minizip/unzip.h>
+#include <minizip/zip.h>
 #if defined(OS_WIN)
 #include "third_party/zlib/contrib/minizip/iowin32.h"
 #endif
--- chromium-browser-5.0.312.0~svn20100201r37680/src/webkit/glue/plugins/plugin_lib_linux.cc~	2010-01-07 23:19:11.000000000 +0200
+++ chromium-browser-5.0.312.0~svn20100201r37680/src/webkit/glue/plugins/plugin_lib_linux.cc	2010-02-02 09:16:08.740140511 +0200
@@ -13,7 +13,7 @@
 
 // These headers must be included in this order to make the declaration gods
 // happy.
-#include "base/third_party/nspr/prcpucfg_linux.h"
+#include <prcpucfg.h>
 
 namespace {
 
--- chromium-browser-5.0.312.0~svn20100201r37680/src/chrome/browser/metrics/metrics_log.cc~	2010-01-20 05:02:33.000000000 +0200
+++ chromium-browser-5.0.312.0~svn20100201r37680/src/chrome/browser/metrics/metrics_log.cc	2010-02-02 09:17:14.496849280 +0200
@@ -12,7 +12,7 @@
 #include "base/scoped_ptr.h"
 #include "base/string_util.h"
 #include "base/sys_info.h"
-#include "base/third_party/nspr/prtime.h"
+#include <prtime.h>
 #include "chrome/browser/autocomplete/autocomplete.h"
 #include "chrome/browser/browser_process.h"
 #include "chrome/common/logging_chrome.h"
