--- chromium-browser-31.0.1650.48/third_party/WebKit/Source/bindings/derived_sources.gyp~	2013-11-08 08:50:07.000000000 +0200
+++ chromium-browser-31.0.1650.48/third_party/WebKit/Source/bindings/derived_sources.gyp	2013-11-13 19:07:02.379771691 +0200
@@ -44,8 +44,6 @@
     ],
     'compiler_module_files': [
         'scripts/idl_compiler.py',
-        '<(DEPTH)/third_party/ply/lex.py',
-        '<(DEPTH)/third_party/ply/yacc.py',
         # jinja2/__init__.py contains version string, so sufficient for package
         '<(DEPTH)/third_party/jinja2/__init__.py',
         '<(DEPTH)/third_party/markupsafe/__init__.py',  # jinja2 dep
