Index: tools/compile_test/compile_test.py
diff --git a/tools/compile_test/compile_test.py b/tools/compile_test/compile_test.py
index 79371a1fd05d2e03e098340527fc897faadad54c..bbda4abfe54f327aabede26f400bebd5e1376d82 100755
--- a/tools/compile_test/compile_test.py
+++ b/tools/compile_test/compile_test.py
@@ -30,7 +30,9 @@ def DoMain(argv):
   if not options.code:
     parser.error('Missing required --code switch.')
 
-  cxx = os.environ.get('CXX', 'g++')
+  # The environment variable might expand to a string with spaces,
+  # e.g. "ccache g++". Convert it to a list suitable for argv.
+  cxx = os.environ.get('CXX', 'g++').split()
 
   tmpdir = tempfile.mkdtemp()
   try:
@@ -41,7 +41,7 @@
 
     o_path = os.path.join(tmpdir, 'test.o')
 
-    cxx_popen = subprocess.Popen([cxx, cxx_path, '-o', o_path, '-c'],
+    cxx_popen = subprocess.Popen(cxx + [cxx_path, '-o', o_path, '-c'],
                                  stdout=subprocess.PIPE,
                                  stderr=subprocess.PIPE)
     cxx_stdout, cxx_stderr = cxx_popen.communicate()
diff --git a/media/filters/ffmpeg_glue.h b/media/filters/ffmpeg_glue.h
index 17241b9..8a92312 100644
--- a/media/filters/ffmpeg_glue.h
+++ b/media/filters/ffmpeg_glue.h
@@ -28,9 +28,9 @@
 #include "base/basictypes.h"
 #include "base/memory/scoped_ptr.h"
 #include "media/base/media_export.h"
+#include "media/ffmpeg/ffmpeg_common.h"
 
 struct AVFormatContext;
-struct AVIOContext;
 
 namespace media {
 
diff --git a/media/media.gyp b/media/media.gyp
index df217d2..fde3830 100644
--- a/media/media.gyp
+++ b/media/media.gyp
@@ -363,6 +363,9 @@
           'dependencies': [
             '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
           ],
+          'export_dependent_settings': [
+            '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
+          ],
         }, {  # media_use_ffmpeg == 0
           # Exclude the sources that depend on ffmpeg.
           'sources!': [
--- a/media/ffmpeg/ffmpeg_common.cc.orig	2013-01-17 00:07:51.635057013 +0000
+++ b/media/ffmpeg/ffmpeg_common.cc	2013-01-17 00:15:50.867406811 +0000
@@ -10,6 +10,8 @@
 #include "media/base/video_frame.h"
 #include "media/base/video_util.h"
 
+#undef SampleFormat
+
 namespace media {
 
 // Why FF_INPUT_BUFFER_PADDING_SIZE? FFmpeg assumes all input buffers are
